cli/src/commands/visualize/calmToDot.ts

```diff
@@ -0,0 +1,128 @@
+import { Digraph, Subgraph, Node, Edge, toDot, attribute } from 'ts-graphviz';
+import { CALMInstantiation, CALMComposedOfRelationship, CALMConnectsRelationship, CALMDeployedInRelationship, CALMInteractsRelationship, CALMRelationship, CALMNode } from './Types';
+import { initLogger } from '../helper.js';
+import winston from 'winston';
+
+let logger: winston.Logger;
+
+const idToNode: {[id: string]: Node} = {};
+
+export default function(calm: CALMInstantiation, debug: boolean = false): string {
+    logger = initLogger(debug);
+
+    const G = new Digraph({
+        nodesep: 0.5
+    });
+
+    calm.nodes.forEach(node => {
+        logger.debug(`Creating a node with ID [${node['unique-id']}]`);
+        addNode(G, node);
+    });
+    calm.relationships.forEach(relationship => {
+        addRelationship(G, relationship);
+    });
+
+    return toDot(G)
+        .replace(/\n/g, ' ');
+}
+
+function addNode(G: Digraph, node: CALMNode) {
+    if (node['node-type'] === 'actor') {
+        createNode(G, node, 'ellipse');
+    } else if (node['node-type'] === 'internal-network') {
+        // do not create node as this will be a subgraph later
+    } else {
+        createNode(G, node, 'box');
+    }
+}
+
+function createNode(G: Digraph, node: CALMNode, shape: string) {
+    const graphNode = new Node(node['unique-id'], {
+        [attribute.shape]: shape,
+        label: `${capitalizeFirstLetter(node['node-type'])}: ${node.name}`
+    });
+    idToNode[node['unique-id']] = graphNode;
+    G.addNode(graphNode);
+}
+
+function capitalizeFirstLetter(s: string): string {
+    return s.charAt(0).toUpperCase() + s.slice(1);
+}
+
+function addRelationship(g: Digraph, relationship: CALMRelationship): void {
+    if ('connects' in relationship['relationship-type']) {
```

Comment:
Could we potentially add methods to work on relationship types such as .getRelationshipType now that we have a model for it?

Reply:
Just a thought, this works fine